keychain/btcwallet.go

```diff
@@ -472,11 +472,18 @@ func (b *BtcWalletKeyRing) SignMessageSchnorr(keyLoc KeyLocator,
 		privKey = txscript.TweakTaprootPrivKey(*privKey, taprootTweak)
 	}
 
+	// If a tag was provided, we need to take the tagged hash of the input.
 	var digest []byte
-	if doubleHash {
-		digest = chainhash.DoubleHashB(msg)
+	if len(tag) == 0 {
+		if doubleHash {
```

Comment:
we should probably error out if `doubleHash` is true _and_ `len(tag)>0`. 

Then we can also turn this into a switch I think